### PR TITLE
fix: remove the `max_uid_per_chain` parameter

### PR DESCRIPTION
**Pull request by tbrezot**

### CODE REVIEW

File: src/entry_table.rs [hunk 1]
```diff
@@ -224,11 +226,10 @@ impl<const UID_LENGTH: usize, const KWI_LENGTH: usize> EntryTableValue<UID_LENGT
     >(
         &self,
         kwi_uid: &KmacKey,
-        max_uid_per_chain: usize,
     ) -> Vec<Uid<UID_LENGTH>> {
         let mut chain = Vec::new();
 
-        while chain.len() < max_uid_per_chain && chain.last() != self.chain_table_uid.as_ref() {
+        while chain.len() < MAX_UID_PER_CHAIN && chain.last() != self.chain_table_uid.as_ref() {
```
Comment (ThibaudDauce) [inline src/entry_table.rs:232]: Should we fail when reaching `MAX_UID_PER_CHAIN`? Here we'll silently return incomplete results, no?

Comment (tbrezot) [reply src/entry_table.rs:232]: indeed, see 7950df0

**Review by ThibaudDauce — APPROVED** [review]

File: src/entry_table.rs [hunk 2]
```diff
@@ -23,6 +23,9 @@ use crate::{
     KeyingMaterial, Keyword, CHAIN_TABLE_KEY_DERIVATION_INFO,
 };
 
+/// Maximum number of Chain Table UID associated to a given Entry Table UID.
+const MAX_UID_PER_CHAIN: usize = 1_000_000;
```
Comment (Manuthor) [inline src/entry_table.rs:27]: I suppose 1_000_000 is a reasonable limit. What happens if this limit is reached at one point? no way to override it right?

Comment (ThibaudDauce) [reply src/entry_table.rs:27]: I think Findex will be unusable way before this limit (timing / size wise). Users shouldn't index so much locations inside the same keyword.

**Pull request merged**